src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -771,6 +772,17 @@ public boolean hasntStartedYet() {
         return checkouts;
     }
 
+    public synchronized @Nullable List<SCM> getCheckouts() {
+       if (checkouts == null) {
```

Comment:
Instead of having this method be `synchronized` and using the `checkouts` field directly, I think you should use `checkouts(TaskListener.null)` instead, then store the result in a local variable and use that in this method so that the Javadoc comment on `checkouts` that says `Access only through checkouts(TaskListener)` stays true and we don't have to worry about synchronization on the same object from multiple methods (even though in this case it is easy to reason about for now).

If you do that, the return value here would be `@Nonnull` since `checkouts(TaskListener)` never returns null, and that seems easier for consumers to handle.